src/outpack/location_pull.py

```diff
@@ -122,16 +121,37 @@ def _mark_all_known(driver, root, location_name):
 def outpack_location_pull_packet(
     ids: Union[str, List[str]],
     *,
-    options=None,
-    recursive=None,
-    root=None,
-    locate=True,
+    options: Optional[SearchOptions] = None,
+    recursive: Optional[bool] = None,
+    root: Union[str, OutpackRoot, None] = None,
+    locate: bool = True,
 ):
     root = root_open(root, locate=locate)
-    options = SearchOptions(options)
+
     if isinstance(ids, str):
         ids = [ids]
-    plan = _location_build_pull_plan(ids, options.location, recursive, root)
+
+    if options is None:
+        actual_options = SearchOptions(allow_remote=True)
+    else:
+        actual_options = SearchOptions.create(options)
+
+    if not actual_options.allow_remote:
+        msg = "'allow_remote' must be True"
+        raise Exception(msg)
+
+    if recursive is None:
+        recursive = root.config.core.require_complete_tree
+
+    if root.config.core.require_complete_tree and not recursive:
```

Comment:
I presume this duplicates the logic in the R version (so this comment applies to both). I wonder if we're being overly cautious here in the case where we are trying to pull packets that don't _need_ a recursive pull?  In that sense it's only really an issue if we work out we need more careful check that we're not needing to pull more than one id?  We can punt this into a ticket for later easily though

Reply:
You mean if the packet we pull either has no dependencies, all the dependencies are already satisfied, or they are also specified as part of `ids`? I think that make sense.